db/entrypoint.bash

```diff
@@ -0,0 +1,12 @@
+#!/usr/bin/env bash
+
+if [ "${DEBUG}" = "true" ]; then
+    set -xEeo pipefail
+fi
+
+if [ -z "${PG_ENCRYPT_KEY}" ]; then
+    echo "\$PG_ENCRYPT_KEY must be set and non-empty."
+    exit 1
+fi
+
+exec /usr/local/bin/docker-entrypoint.sh postgres -c encrypt.key="${PG_ENCRYPT_KEY}" "$@"
```

Comment:
The base entrypoint here makes this a bit specific to the base image used. I wonder if we could add something in the Dockerfile like create a symlink to a base image agnostic entrypoint like `/usr/local/bin/postgres-entrypoint` and then both the upstream and downstream images could use this script as-is. This would be handy in case the script evolves over time.

Reply:
I'm not sure I follow - what is `/usr/local/bin/postgres-entrypoint` in this example, or what would it be?

I think this setup is still going to be fragile to base image changes because the `/docker-entrypoint-initdb.d/` script loading mechanism is a feature of the `docker-entrypoint.sh`, too. If the base image doesn't have that same entrypoint then the mechanism for enabling the pgcrypto extension would probably also need to be changed.

Reply:
I guess what I'm asking for is just `RUN ln -s /usr/local/bin/docker-entrypoint /usr/local/bin/postgres-entrypoint`, and then reference `postgres-entrypoint` in the Bash script. Then we can have a similar symlink in the downstream Dockerfile based on `rhel8/postgres-15`. That image does have a similar init directory for scripts thankfully.

The Dockerfiles will have to be slightly different at least, I just would like the Bash scripts and any other files to be reusable between upstream/downstream.